File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/util/MessageSender.java
package uk.gov.hmcts.reform.bulkscan.orchestrator.util;

import com.microsoft.azure.servicebus.IMessageHandler;
import com.microsoft.azure.servicebus.Message;

public class MessageSender {

    private final IMessageHandler processor;

    public MessageSender(IMessageHandler processor) {
        this.processor = processor;
    }

    public void send(Message message) {
        processor.onMessageAsync(message).join();
    }
}

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/util/MessageSender.kt
package uk.gov.hmcts.reform.bulkscan.orchestrator.util

import com.microsoft.azure.servicebus.IMessageHandler
import com.microsoft.azure.servicebus.Message
import org.springframework.beans.factory.annotation.Autowired

class MessageSender(@Autowired private val processor: IMessageHandler) {
    fun send(message: Message) {
        processor.onMessageAsync(message).get()
    }
}
